gson/src/test/java/com/google/gson/functional/Java17RecordTest.java

```diff
@@ -67,21 +175,201 @@ public void testPrimitiveDefaultValues() {
   }
 
   @Test
-  public void testPrimitiveNullValues() {
-    RecordWithPrimitives expected = new RecordWithPrimitives("s", (byte) 0, (short) 0, 0, 0, 0, 0, '\0', false);
-    // TODO(eamonnmcmanus): consider forbidding null for primitives
-    String s = "{'aString': 's', 'aByte': null, 'aShort': null, 'anInt': null, 'aLong': null, 'aFloat': null, 'aDouble': null, 'aChar': null, 'aBoolean': null}";
-    assertEquals(expected, gson.fromJson(s, RecordWithPrimitives.class));
+  public void testPrimitiveJsonNullValue() {
+    String s = "{'aString': 's', 'aByte': null, 'aShort': 0}";
+    var e = assertThrows(JsonParseException.class, () -> gson.fromJson(s, RecordWithPrimitives.class));
+    assertEquals("null is not allowed as value for record component 'aByte' of primitive type; at path $.aByte",
+        e.getMessage());
   }
 
-  public record MyRecord(
-      @SerializedName("name") String a,
-      @SerializedName(value = "name1", alternate = {"name2", "name3"}) String b) {
-    public MyRecord {
-      a = "modified-" + a;
-    }
+  /**
+   * Tests behavior when JSON contains non-null value, but custom adapter returns null
+   * for primitive component
+   */
+  @Test
+  public void testPrimitiveAdapterNullValue() {
+    Gson gson = new GsonBuilder()
+        .registerTypeAdapter(byte.class, new TypeAdapter<Byte>() {
+          @Override public Byte read(JsonReader in) throws IOException {
+            in.skipValue();
+            // Always return null
+            return null;
+          }
+
+          @Override public void write(JsonWriter out, Byte value) {
+            throw new AssertionError("not needed for test");
+          }
+        })
+        .create();
+
+    String s = "{'aString': 's', 'aByte': 0}";
+    var exception = assertThrows(JsonParseException.class, () -> gson.fromJson(s, RecordWithPrimitives.class));
+    assertEquals("null is not allowed as value for record component 'aByte' of primitive type; at path $.aByte",
+        exception.getMessage());
   }
 
-  public record RecordWithPrimitives(
+  private record RecordWithPrimitives(
       String aString, byte aByte, short aShort, int anInt, long aLong, float aFloat, double aDouble, char aChar, boolean aBoolean) {}
+
+  /** Tests behavior when value of Object component is missing; should default to null */
+  @Test
+  public void testObjectDefaultValue() {
+    record LocalRecord(String s, int i) {}
+
+    assertEquals(new LocalRecord(null, 1), gson.fromJson("{\"i\":1}", LocalRecord.class));
+  }
+
+  /**
+   * Tests serialization of a record with {@code static} field.
+   *
+   * <p>Important: It is not documented that this is officially supported; this
+   * test just checks the current behavior.
+   */
+  @Test
+  public void testStaticFieldSerialization() {
+    Gson gson = new GsonBuilder()
+        // Include static fields
+        .excludeFieldsWithModifiers(0)
```

Comment:
Could you also test that the field is _not_ serialized without this?

Reply:
It looks like there was already a test for this: `com.google.gson.functional.UncategorizedTest.testStaticFieldsAreNotSerialized()`
And additionally multiple tests implicitly relied on `static` fields being ignored by default. However, I have now extended these new test methods to also test the default Gson behavior.

Also, I have noticed that `Field.set` per documentation does not support `static final` fields. So to avoid Gson claiming "Unexpected IllegalAccessException occurred...", I have added custom handling for that case to provide a better exception message.